packages/features/ee/impersonation/lib/ImpersonationProvider.ts

```diff
@@ -63,11 +64,75 @@ export function checkUserIdentifier(creds: Partial<Credentials>) {
 }
 
 export function checkPermission(session: Session | null) {
-  if (session?.user.role !== "ADMIN" && process.env.NEXT_PUBLIC_TEAM_IMPERSONATION === "false") {
+  if (
+    (session?.user.role !== "ADMIN" && process.env.NEXT_PUBLIC_TEAM_IMPERSONATION === "false") ||
```

Comment:
probably a good spot to use `!Boolean(process.env.NEXT_PUBLIC_TEAM_IMPERSONATION)` instead of `process.env.NEXT_PUBLIC_TEAM_IMPERSONATION === "false"`

because: Boolean("true") => true and  Boolean("false") = false and Boolean(undefined) = false